src/client/builtin_scene/shaders/gaussian_splatting.vert

```diff
@@ -0,0 +1,257 @@
+vec3 quatVec(vec4 q, vec3 v)
+{
+  vec3 t = 2.0 * cross(q.xyz, v);
+  return v + q.w * t + cross(q.xyz, t);
+}
+
+vec4 quatQuat(vec4 q1, vec4 q2)
+{
+  return vec4(q1.w * q2.x + q1.x * q2.w + q1.y * q2.z - q1.z * q2.y, q1.w * q2.y - q1.x * q2.z + q1.y * q2.w + q1.z * q2.x, q1.w * q2.z + q1.x * q2.y - q1.y * q2.x + q1.z * q2.w, q1.w * q2.w - q1.x * q2.x - q1.y * q2.y - q1.z * q2.z);
+}
+
+mat3 scaleQuaternionToMatrix(vec3 s, vec4 q)
+{
+  return mat3(s.x * (1.0 - 2.0 * (q.y * q.y + q.z * q.z)), s.x * (2.0 * (q.x * q.y + q.w * q.z)), s.x * (2.0 * (q.x * q.z - q.w * q.y)), s.y * (2.0 * (q.x * q.y - q.w * q.z)), s.y * (1.0 - 2.0 * (q.x * q.x + q.z * q.z)), s.y * (2.0 * (q.y * q.z + q.w * q.x)), s.z * (2.0 * (q.x * q.z + q.w * q.y)), s.z * (2.0 * (q.y * q.z - q.w * q.x)), s.z * (1.0 - 2.0 * (q.x * q.x + q.y * q.y)));
+}
+
+vec4 mat3ToQuat(mat3 m)
+{
+  float trace = m[0][0] + m[1][1] + m[2][2];
+  vec4 q;
+
+  if (trace > 0.0)
+  {
+    float s = sqrt(trace + 1.0) * 2.0;
+    q.w = 0.25 * s;
+    q.x = (m[2][1] - m[1][2]) / s;
+    q.y = (m[0][2] - m[2][0]) / s;
+    q.z = (m[1][0] - m[0][1]) / s;
+  }
+  else if ((m[0][0] > m[1][1]) && (m[0][0] > m[2][2]))
+  {
+    float s = sqrt(1.0 + m[0][0] - m[1][1] - m[2][2]) * 2.0;
+    q.w = (m[2][1] - m[1][2]) / s;
+    q.x = 0.25 * s;
+    q.y = (m[0][1] + m[1][0]) / s;
+    q.z = (m[0][2] + m[2][0]) / s;
+  }
+  else if (m[1][1] > m[2][2])
+  {
+    float s = sqrt(1.0 + m[1][1] - m[0][0] - m[2][2]) * 2.0;
+    q.w = (m[0][2] - m[2][0]) / s;
+    q.x = (m[0][1] + m[1][0]) / s;
+    q.y = 0.25 * s;
+    q.z = (m[1][2] + m[2][1]) / s;
+  }
+  else
+  {
+    float s = sqrt(1.0 + m[2][2] - m[0][0] - m[1][1]) * 2.0;
+    q.w = (m[1][0] - m[0][1]) / s;
+    q.x = (m[0][2] + m[2][0]) / s;
+    q.y = (m[1][2] + m[2][1]) / s;
+    q.z = 0.25 * s;
+  }
+  return normalize(q);
+}
+
+void decomposeViewMatrix(
+  mat4 viewMatrix,
+  out vec3 viewPosition,
+  out vec4 viewQuaternion
+)
+{
+  mat3 viewRotMatrix = mat3(viewMatrix);
+  mat3 worldRotMatrix = transpose(viewRotMatrix);
+  float det = determinant(viewRotMatrix);
+
+  if (det > 0.0)
+  {
+    viewQuaternion = mat3ToQuat(worldRotMatrix);
+  }
+  else
+  {
+    worldRotMatrix[2] = -worldRotMatrix[2];
+    viewQuaternion = mat3ToQuat(worldRotMatrix);
+  }
+
+  viewPosition = viewMatrix[3].xyz;
+}
+
+void main()
+{
+  // Default to outside the frustum so it's discarded if we return early
+  gl_Position = vec4(0.0, 0.0, 2.0, 1.0);
+
+    // Handle special value for "no splat"
+  if (splatIndex == 0xffffffffu)
+  {
+    return;
+  }
+
+  // Get texture coordinate using efficient bit operations
+  ivec2 texCoord = getSplatTexCoord(int(splatIndex));
+
+  // Direct texture fetches - no pack/unpack needed!
+  vec3 center = texelFetch(splatCenters, texCoord, 0).rgb;
+  vec4 rgba = texelFetch(splatColors, texCoord, 0);
+  vec3 scales = texelFetch(splatScales, texCoord, 0).xyz;
+  vec4 quaternion = texelFetch(splatQuat, texCoord, 0);
+
```

Comment:
Add a new `uniform float maxDistance`, which is used to return directly at this line when the distance from current center to camera is larger than the value, this might help to address the GPU overdraw issue.

And the default `maxDistance` is 2 (meters).